share/build-navigation/build-navigation-json.xsl

```diff
@@ -108,8 +114,162 @@
+        <xsl:call-template name="lifecycle-json">
+          <!-- Include also <product>s which don't have a site-section attribute
+               Such <product> elements are considered the same as
+               <product site-section="$default_site_section">
+          -->
+          <xsl:with-param name="nodes" select="product[@site-section=$single_site]
+                                               |product[not(@site-section)]"/>
+          <xsl:with-param name="lifecycle">supported</xsl:with-param>
+          <xsl:with-param name="site" select="$single_site"/>
+        </xsl:call-template>
+
+        <xsl:call-template name="lifecycle-json">
+          <xsl:with-param name="nodes" select="product[@site-section=$single_site]
+                                               |product[not(@site-section)]"/>
+          <xsl:with-param name="lifecycle">unsupported</xsl:with-param>
+          <xsl:with-param name="site" select="$single_site"/>
+        </xsl:call-template>
+      </xsl:when>
+      <xsl:otherwise>
+        <xsl:call-template name="lifecycle-json">
+          <xsl:with-param name="nodes" select="product[@site-section=$single_site]"/>
+          <xsl:with-param name="lifecycle">supported</xsl:with-param>
+          <xsl:with-param name="site" select="$single_site"/>
+        </xsl:call-template>
+
+        <xsl:call-template name="lifecycle-json">
+          <xsl:with-param name="nodes" select="product[@site-section=$single_site]"/>
+          <xsl:with-param name="lifecycle">unsupported</xsl:with-param>
+          <xsl:with-param name="site" select="$single_site"/>
+        </xsl:call-template>
+      </xsl:otherwise>
+    </xsl:choose>
+  </xsl:template>
+
+
+  <xsl:template name="lifecycle-json">
+    <xsl:param name="nodes"/>
+    <xsl:param name="lifecycle"/>
+    <xsl:param name="site"/>
+    <xsl:variable name="filename" select="concat($output_root, $site, '-', $lifecycle, '.json')"/>
+    <xsl:variable name="list">
+      <xsl:choose>
+        <xsl:when test="$lifecycle='supported'">
+          <xsl:value-of select="$existing-sets-supported"/>
+        </xsl:when>
+        <xsl:otherwise>
+          <xsl:value-of select="$existing-sets-unsupported"/>
+        </xsl:otherwise>
+      </xsl:choose>
+    </xsl:variable>
+
+   <xsl:message>lifecycle-json:
+     JSON file="<xsl:value-of select="$filename"/>"
+     nodes=<xsl:value-of select="count($nodes)"/>
+     all product nodes=<xsl:value-of select="count(//product)"/>
+     current node=<xsl:value-of select="local-name(.)"/>
+     list=<xsl:value-of select="$list"/>
+   </xsl:message>
+
+    <exsl:document
+      href="{$filename}"
+      method="text"
+      encoding="UTF-8"
+      indent="no"
+      media-type="application/x-json">
+{
+  "site-section": "<xsl:value-of select="$site"/>",
+  "productline": {
+     <xsl:apply-templates select="$nodes[docset/@lifecycle=$lifecycle]" mode="generate-productline-list">
```

Comment:
Checking for `@lifecycle=$lifecycle` is not really correct -- see `relevant-docs.xsl` which has this logic:

```
((
               $list = 'supported' and
               (@lifecycle = 'beta' or @lifecycle = 'supported' or
               (@lifecycle = 'unpublished' and $internal_mode = 'true'))
              ) or (
               $list = 'unsupported' and
               @lifecycle = 'unsupported'
              )) and
              (not(@navigation = 'hidden') or $internal_mode = 'true') and
              not(@navigation = 'disabled')
```

Reply:
This is already fixed in commit 5035bb0af